process_test.go

```diff
@@ -166,3 +173,171 @@ func TestProcessImplementations(t *testing.T) {
 		})
 	}
 }
+
+// Returns path to release and to mongod
+func downloadMongodb(t *testing.T) (string, string) {
+	var target string
+	var edition string
+	platform := runtime.GOOS
+	switch platform {
+	case "darwin":
+		target = "osx"
+		edition = "enterprise"
+	case "linux":
+		edition = "base"
+		target = platform
+	default:
+		edition = "enterprise"
+		target = platform
+	}
+	arch := bond.MongoDBArch("x86_64")
+	release := "4.0-stable"
+
+	dir, err := ioutil.TempDir("build", "mongodb")
+	require.NoError(t, err)
+
+	opts := bond.BuildOptions{
+		Target:  target,
+		Arch:    bond.MongoDBArch(arch),
+		Edition: bond.MongoDBEdition(edition),
+		Debug:   false,
+	}
+	releases := []string{release}
+	require.NoError(t, recall.DownloadReleases(releases, dir, opts))
+
+	catalog, err := bond.NewCatalog(dir)
+	require.NoError(t, err)
+
+	path, err := catalog.Get("4.0-current", string(edition), target, string(arch), false)
+	require.NoError(t, err)
+
+	var mongodPath string
+	if platform == "windows" {
+		mongodPath = filepath.Join(path, "bin", "mongod.exe")
+	} else {
+		mongodPath = filepath.Join(path, "bin", "mongod")
+	}
+	_, err = os.Stat(mongodPath)
+	require.NoError(t, err)
+
+	return dir, mongodPath
+}
+
+func setupMongods(numProcs int, mongodPath string) ([]CreateOptions, []string, error) {
+	dbPaths := make([]string, numProcs)
+	optslist := make([]CreateOptions, numProcs)
+	for i := 0; i < numProcs; i++ {
+		procName := "mongod"
+		port := getPortNumber()
+
+		dbPath, err := ioutil.TempDir("build", procName)
+		if err != nil {
+			return nil, nil, err
+		}
+		dbPaths[i] = dbPath
+
+		opts := CreateOptions{Args: []string{mongodPath, "--port", fmt.Sprintf("%d", port), "--dbpath", dbPath}}
+		optslist[i] = opts
+	}
+
+	return optslist, dbPaths, nil
+}
+
+func teardownMongods(dbPaths []string) {
```

Comment:
this function should probably be "removeDBPaths"

Reply:
Done